Fix build

mattdean-digicatapult opened the pull request:
Exclude tests from build output to fix build following #19.  

Also move types into src to avoid generation of `scenario.js` on build which could end up out of sync. Note this is a rather opinionated change and I'm not sure why types was top level to start with. Happy to have a conversation on this point if I've missed something.

mattdean-digicatapult committed:
Exclude tests from build output to fix build. Also move types into src to avoid generation of scenario.js on build which could end up out of sync

mattdean-digicatapult committed:
Add build to test workflow

mattdean-digicatapult committed:
Add further excludes in case we add non-test .ts files as helpers to the test directories

n3op2 reviewed (approved):
Looks like just types have been moved into the src/ directory. Does all npm scripts work?

mattdean-digicatapult commented:
There are three main changes:

1. Added to test workflow a run of `npm run build`. This was what was missed on #19 
2. Added `excludes` to `tsconfig.json`. This prevents certain files from being considered part of the build output. Notably this is the tests which we wouldn't want to distribute with a build so that's fine. This is the actual fix.
3. Move `types` into `src`. The main reason for this is the `types` are semantically part of the program we want to build so they should be in the `buildRoot` as configured in `tsconfig.json`. Note this resolves an issue where the compiler was compiling the types separately to the program and generating `scenario.js`. Note that the only reason this worked is because `build` and `src` are the same level of indirection away from `types`. If we had built a container with only build I don't think this would have worked
